modules/stochastic_tools/src/userobject/SurrogateModel.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SurrogateModel.h"
+
+defineLegacyParams(SurrogateModel);
+
+InputParameters
+SurrogateModel::validParams()
+{
+  InputParameters params = GeneralUserObject::validParams();
+  params += SamplerInterface::validParams();
+
+  params.addRequiredParam<SamplerName>("training_sampler",
```

Comment:
Just use "sampler", that is what I did for a bunch of other objects.

Reply:
I think we want to distinguish between `training_sampler` and `sampler`. One sampler to train the surrogate and another to actually use it. We don't yet have the infrastructure on the latter, but I figured this would be a good place to start.

Reply:
OK.